### ISSUE TITLE
Rewrite CMS tests

### ISSUE THREAD

**Issue by dominicfollett** [proversity-org/edx-platform#69]
For ORGANIZATION_APP feature setting.

**Comment by dominicfollett** [comment]
Create Organization factory, with default course short_name setting, if Feature setting is set, create Organization factory object, otherwise leave it.

**Comment by dominicfollett** [comment]
Picked up issue with admin being affected by org filtering middleware. Need to make sure this middleware ONLY affects the Studio, NOT admin, NOT LMS or anything else.
Besides that, this issue arose when I tested creating two identical organizations with the exact same data. -- This should not be allowed either.

**Comment by dominicfollett** [comment]
With some tests, when ORGANIZATION_APP is true, their tests do not account for the fact in this scenario there is ALWAYS a check made first to see if the org exists, regardless of whether the ORG name is invalid (has forbidden characters etc). This is also results in test failures. It doesn't make sense to add an ORG with an invalid name, as this should not be allowed by default. Better would be to change the ordering of these executions so that invalid ORG name is caught first before anything else.

**Comment by dominicfollett** [comment]
As a sanity check, run tests again with ORGANIZATION_APP set to false -- these should not fail either.

**Comment by dominicfollett** [comment]
Note I've changed the test:  `cms/djangoapps/contentstore/tests/test_contentstore.py:RerunCourseTest.test_rerun_course_wiki_slug`

so that it will conform to our way of preventing a user who is part of ORG x from rerunning or creating a course under a different ORG. 

It appears Edx was happy for a user to be able to do this across organizations.
This time I've ensured that the rerun course is done under the same ORG and left the rest of the test semantics the same.

However I might want to just check the case  where the creator is not affiliated with an ORG, then creating accross ORGS should be allowed. As in the instance where we, Proversity, go in as admins and perform some course creation/rerun task.
